### PR TITLE
dotCMS/core#11513 Develop create/edit content type item

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/app/api/services/crud-service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { fakeAsync, tick } from '@angular/core/testing';
```
Comment (fmontes) [inline src/app/api/services/crud-service.spec.ts]: Just rename the service.

File: src/app/api/services/site-service.ts [hunk 2]
```diff
@@ -64,6 +64,10 @@ export class SiteService {
         return this.selectedSite;
     }
 
+    get loadedSites(): Site[] {
```
Comment (fmontes) [inline src/app/api/services/site-service.ts]: This will probably change after we fix the issue in the backend.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,16 @@
+root = true
```
Comment (fmontes) [inline .editorconfig]: http://editorconfig.org/

File: src/app/portlets/content-types-create-edit/content-types-form/content-types-form.spec.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        comp.form.controls.urlMapPattern.setValue('pattern');
+        comp.form.controls.workflow.setValue('workflow-identifier');
+        comp.form.controls.host.setValue('host-identifier');
+        comp.submitContent();
+        expect(crudService.postData).toHaveBeenCalledWith('v1/contenttype', mockData);
+        expect(comp.readyToAddFields).toBeTruthy();
+        expect(comp.submitAttempt).toBeFalsy();
+    });
+
+    xit('should have full form collapsed by default', () => {
```
Comment (freddyucv) [inline src/app/portlets/content-types-create-edit/content-types-form/content-types-form.spec.ts]: Add TODO here

File: src/app/portlets/content-types-create-edit/content-types-form/content-types-form.spec.ts [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        // TODO: Needs to figute it out why by default the offsetHeight it's not 0
+        fixture.detectChanges();
+        let fullForm: DebugElement = fixture.debugElement.query(By.css('#content-type-form-full'));
+        let fullFormEl: HTMLElement = fullForm.nativeElement;
+        expect(fullFormEl.offsetHeight).toBe(0);
+    });
+
+    xit('should expand full form action button click', () => {
```
Comment (freddyucv) [inline src/app/portlets/content-types-create-edit/content-types-form/content-types-form.spec.ts]: Add TODO here

File: src/app/portlets/content-types-create-edit/content-types-form/content-types-form.ts [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+            }
+        ];
+
+        this.expireDateFieldOptions = [
+            {
+                label: 'Select one',
+                value: null
+            }
+        ];
+    }
```
Comment (freddyucv) [inline src/app/portlets/content-types-create-edit/content-types-form/content-types-form.ts]: too long method

File: src/app/portlets/content-types-create-edit/content-types-form/content-types-form.html [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+              </div>
+          </div>
+          <div class="form__buttons">
+            <button pButton type="button" flat id="content-type-form-cancel" label="{{i18nMessages['cancel']}}" (click)="goToListing()"></button>
+            <button pButton type="submit" id="content-type-form-submit" label="{{i18nMessages['save']}}"></button>
+          </div>
+        </div>
+      </form>
+      <div *ngIf="readyToAddFields">
+        <h1>Now you can add fields</h1>
```
Comment (fmontes) [inline src/app/portlets/content-types-create-edit/content-types-form/content-types-form.html]: This is a placeholder text, will not exist after we add the drag and drop functionality for the fields

File: src/app/portlets/content-types-create-edit/content-types-form/content-types-form.html [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+      </div>
+    </div>
+  </div>
+  <div class="content-type__fields-sidebar">
+    <p-tabView>
+      <p-tabPanel header="{{i18nMessages['fields']}}">
+        <div class="container-fluid">
+          Fields
```
Comment (fmontes) [inline src/app/portlets/content-types-create-edit/content-types-form/content-types-form.html]: Placeholders

File: src/app/portlets/content-types-create-edit/content-types-form/content-types-form.html [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+        </div>
+      </p-tabPanel>
+      <p-tabPanel header="{{i18nMessages['Properties']}}">
+        <div class="container-fluid">
+          Properties
```
Comment (fmontes) [inline src/app/portlets/content-types-create-edit/content-types-form/content-types-form.html]: Placeholders

File: src/app/portlets/content-types-create-edit/content-types-form/content-types-form.spec.ts (hunk 4 above)
Comment (fmontes) [inline src/app/portlets/content-types-create-edit/content-types-form/content-types-form.spec.ts]: There is literally a TODO in the next line.

**Pull request merged**